File: 프로그래머스/lv0/120841. 점의 위치 구하기/점의 위치 구하기.js
function solution(dot) {
//x=dot[0] , y=dot[1]
    if(dot[0] > 0 && dot[1] >0){
        return 1;
    }else if(dot[0] < 0 && dot[1] > 0){
        return 2;
    }else if(dot[0] < 0 && dot[1] < 0){
        return 3;
    }else {
        return 4;
    }
    
}


   
//이중삼항연사자를 사용한 풀이
function solution(dot) {
 
    return dot[0] > 0? (dot[1] > 0 ? 1 : 4) : (dot[1] > 0 ? 2 : 3);
    // dot[0] > 0 일때(true), 즉 x좌표가 양수일때 dot[1](y좌표)의 값이 0보다 크면 1사분면, 아니면 4사분면
    // dot[0] > 0 아닐때(false), x좌표가 음수일때 dot[1](y좌표)의 값이 0보다 작으면 2사분면, 아니면 3사분면 
}

File: 프로그래머스/lv0/120841. 점의 위치 구하기/점의 위치 구하기.java
//방법1
class Solution {
    public int solution(int[] dot) {
        int answer = 0;
        
        if(dot[0]>0 && dot[1]>0){
            return answer=1;
        }else if(dot[0]<0 && dot[1]>0){
            return answer=2;
        }else if(dot[0]<0 && dot[1]<0){
            return answer=3;
        }else if(dot[0]>0 && dot[1]<0){
            return answer=4;
        }
        
        return answer;
    }
}


//방법2
//class Solution {
//    public int solution(int[] dot) {
        
//        if(dot[0]>0){
//            if(dot[1]>0){
//                return 1;
//            }else{
//                return 4;
//            }
//        }else{
//            if(dot[1]>0){
//                return 2;
//            }else{
//               return 3;
//            }
//        }
        
//    }
//}